### DISCUSSION TITLE
Seeking Developer Guidance: Enhancing Vendure for LMS Capabilities with Multi-Media Course Support

### DISCUSSION THREAD

**Discussion by hisk** [vendurehq/vendure#2660] [Q&A]
Hello,

First of all, congratulations for developing Vendure, it is a well rounded project that I enjoy watching being built in the open!

I am exploring the expansion of Vendure to function as a Learning Management System (LMS), enabling store owners to create and manage online courses. While I have a grasp on implementing features for simple digital products with downloadable URLs, I seek guidance on steps required to implement a more comprehensive solution. Specifically, I aim to support the creation of digital products (courses) that encompass various media types, including text, video, and quizzes. What would be your recommended approach for implementing each media type within a unified product type? I'm interested in a developer-focused perspective on the necessary features, considering the complexities involved.

Thank you!

**Comment by michaelbromley** [comment]
Hi there,

This is quite a broad question, and kinda hard for me to answer without spending a long time thinking about what your requirements might be and how best to model them.

I think I'd be more able to help if you detail how _you_ are considering approaching the task, and I can comment on any issues I can see with your plans. Also, if you would like to join our Discord community (https://vendure.io/community) you may be able to get additional feedback from other community members.
